processor/src/range/aux_trace.rs

```diff
@@ -200,16 +200,19 @@ impl AuxTraceBuilder {
             .take(main_trace.num_rows() - NUM_RAND_ROWS)
             .skip(self.start_16bit)
         {
-            p1_idx = row_idx + 1;
+            p1_idx = row_idx as u32 + 1;
 
-            p1[p1_idx] = p1[row_idx] * hint.to_value(*lookup, alphas);
+            p1[p1_idx as usize] = p1[row_idx] * hint.to_value(*lookup, alphas);
 
-            if let Some(range_check) = self.cycle_range_checks.get(&row_idx) {
+            // this is not beautiful, but we need to borrow row_idx as u32 in the cycle_range_checks function
+            let row_idx_u32 = row_idx as u32;
+
+            if let Some(range_check) = self.cycle_range_checks.get(&row_idx_u32) {
```

Comment:
I think you should be able to avoid this by doing the following:

```
            if let Some(range_check) = self.cycle_range_checks.get(&(row_idx as u32)) {
```

Reply:
Ah I didn't know that this is possible. Nice